classes/Video.php

```diff
@@ -613,12 +613,22 @@ public function getHttpResponse(array $headers = [])
     {
         $client = new Client();
         $urls = $this->getUrl();
+        $stream_context_options = [];
+
+        if (strpos($this->webpageUrl, 'xhamster.com') !== false) {
```

Comment:
I would prefer a more generic solution (other websites might require a referer).
You can check if there is a `Referer` key in `$this->http_headers` and reuse that.

Reply:
I understand your concerns.
Something like this should work in a more general way.